### PR TITLE
fix(sequencers): permissioned address technical debt

### LINKED ISSUE #396: allow whitelist techincal debt

### PR DESCRIPTION
**Pull request by trinitys7**
Closes #396

### CODE REVIEW

File: x/denommetadata/keeper/keeper.go [hunk 1]
```diff
@@ -49,3 +52,19 @@ func (k *Keeper) SetHooks(sh types.MultiDenomMetadataHooks) {
...
+func (k *Keeper) isAddressPermissioned(ctx sdk.Context, address string) bool {
+	logger := k.Logger(ctx)
+	accAddr, err := sdk.AccAddressFromBech32(address)
+	if err != nil {
+		logger.Error("failed to extract account address from bech32: ", err)
+		return false
+	}
+
+	return k.sequencerKeeper.HasPermission(ctx, accAddr, types.ModuleName)
+}
```
[the remark this answers is not in the document]
Comment (danwt) [reply x/denommetadata/keeper/keeper.go:70]: or use must and panic

File: x/sequencers/proposal_handler.go [hunk 2]
```diff
@@ -13,45 +13,48 @@ func NewUpdatePermissionProposalHandler(k *keeper.Keeper) govtypes.Handler {
...
 		accAddr, err := sdk.AccAddressFromBech32(addrPerms.Address)
 		if err != nil {
 			return err
 		}
 
-		k.RevokePermissions(ctx, accAddr, addrPerms.PermissionList)
+		action(ctx, accAddr, addrPerms.PermissionList)
 	}
 	return nil
 }
```
[the remark this answers is not in the document]
Comment (danwt) [reply x/sequencers/proposal_handler.go:60]:
```
func NewUpdatePermissionProposalHandler(k *keeper.Keeper) govtypes.Handler {
	return func(ctx sdk.Context, content govtypes.Content) error {
		if err := content.ValidateBasic(); err != nil {
			return err
		}
		switch c := content.(type) {
		case *types.GrantPermissionsProposal:
			return HandlePermissionsProposal(ctx, k.GrantPermissions, c.AddressPermissions)
		case *types.RevokePermissionsProposal:
			return HandlePermissionsProposal(ctx, k.RevokePermissions, c.AddressPermissions)
		default:
			return errorsmod.Wrapf(sdkerrors.ErrUnknownRequest, "unrecognized permissions proposal content type: %T", c)
		}
	}
}

// HandlePermissionsProposal is a handler for executing a permissions proposal
func HandlePermissionsProposal(ctx sdk.Context, fn func(ctx sdk.Context, accAddr sdk.AccAddress, p types.PermissionList), perms []types.AddressPermissions) error {
	for _, addrPerms := range perms {
		accAddr, err := sdk.AccAddressFromBech32(addrPerms.Address)
		if err != nil {
			return err
		}

		fn(ctx, accAddr, addrPerms.PermissionList)
	}
	return nil
}
```

credit @zale144

File: x/sequencers/types/permission.go [hunk 3]
```diff
@@ -24,19 +24,15 @@ func (p *PermissionList) Validate() error {
 	if len(p.Permissions) == 0 {
 		return errors.New("permissions field cannot be empty")
 	}
```
Comment (danwt) [inline x/sequencers/types/permission.go:26]: would still suggest removing this
empty permissions is fine

Comment (trinitys7) [reply x/sequencers/types/permission.go:26]: @danwt My own opinion, if the permission is empty, we should not set it to kv store. I also made a check in RevokePermission to delete the empty permission address.

**Comment by omritoptix** [comment]
gonna merge it for now even if there are open commetns as they are not blockers.

**Review by omritoptix — APPROVED** [review]

**Pull request merged**